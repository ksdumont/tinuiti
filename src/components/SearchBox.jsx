import React, { useState } from 'react';
import PropTypes from 'prop-types';

function SearchBox({ getProducts }) {
  const [text, setText] = useState('');

  const onChange = (e) => {
    setText(e);
    getProducts(text);
  };
  console.log(text);
  return (
    <section className="search">
      <form>
        <input
          type="text"
          className="form-control"
          placeholder="Search Products"
          value={text}
          onChange={(e) => onChange(e.target.value)}
        />
      </form>
    </section>
  );
}

SearchBox.propTypes = {
  getProducts: PropTypes.func,
};
SearchBox.defaultProps = {
  getProducts: () => {},
};

export default SearchBox;
