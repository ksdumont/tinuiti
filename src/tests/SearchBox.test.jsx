import React from 'react';
import ReactDOM from 'react-dom';
import SearchBox from '../components/SearchBox';

import Enzyme, { shallow, render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

it('renders correctly', () => {
  const wrapper = shallow(<SearchBox />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
