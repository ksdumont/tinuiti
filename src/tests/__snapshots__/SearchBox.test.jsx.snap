// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="search"
>
  <form>
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Search Products"
      type="text"
      value=""
    />
  </form>
</section>
`;
